app/assets/javascripts/views/common/tableView.js

```diff
@@ -0,0 +1,174 @@
+(function(App) {
+
+  'use strict';
+
+
+  App.View.TableView = Backbone.View.extend({
+
+    defaults: {
+      resultsPerPage: 3
+    },
+
+    el: '#table',
+
+    events: {
+      'click .pagination-link': '_updatePagination',
+      'change #filter-sort' : '_onSortResults',
+      'keyup #filter-search' : '_onFilterResults'
+    },
+
+    initialize: function(settings) {
+      this.options = {
+        headers: settings.headers,
+        resultsPerPage: settings.resultsPerPage || this.defaults.resultsPerPage,
+        tab: settings.tab,
+        isSortable: settings.filters.sortBy ? settings.filters.sortBy : false,
+        hasSearch: settings.filters.search ? settings.filters.search : false
+      };
+
+      this.paginationIndex = 0;
+
+      this.template = HandlebarsTemplates['common/table'];
+
+      this._initCollection();
+      this._setListeners();
+      this._setVars();
+
+      this._renderFilters();
+
+      this.collection.fetch();
+    },
+
+    _initCollection: function() {
+      var tab = this.options.tab,
+        collectionOptions = {
+          headers : this.options.headers,
+          resultsPerPage: this.options.resultsPerPage
+        };
+
+      if(tab == 'sites') {
```

Comment:
This bit needs a bit more of a thinking. If this view is a reusable component, it shouldn't be aware of the data it will display.